platform/src/main/java/org/hillview/sketches/Histogram.java

```diff
@@ -32,6 +33,35 @@ public Histogram(final IHistogramBuckets bucketDescription) {
         this.buckets = new long[bucketDescription.getNumOfBuckets()];
     }
 
+    // Note that we ceil in order to round up to the nearest leaf
```

Comment:
I think you should probably extend the Histogram class with a PrivateHistogram class and add these methods there.

Reply:
If I subclass Histogram, it seems like I can't use the HistogramSketch directly, as it returns a Histogram rather than the subclass (and slices away the relevant postprocessing method and attributes). Creating a new PrivateHistogramSketch, etc., seems like it will involve duplicating a lot of code. Any suggestions on a better way to do this?

Reply:
Do you still need these methods here?
I see that they are already in the PrivateHistogram.